tests/templates/test_integration.py

```diff
@@ -176,558 +138,331 @@
-        qml.PauliX(wires=1)
-        return [qml.expval(qml.Identity(0)), qml.expval(qml.PauliX(1))]
-
-    return circuit
-
-
-def qnode_cv_kwargs(dev, interface, template1, template2):
-    """CV qnode factory passing differentiable parameters as keyword arguments"""
-
-    @qml.qnode(dev, interface=interface)
-    def circuit(nondiffable1=None, nondiffable2=None):
-        # Add wires
-        nondiffable1['wires'] = range(2)
-        nondiffable2['wires'] = range(2)
-
-        # Circuit
-        qml.Displacement(1., 1., wires=0)
-        template1(**nondiffable1)
-        template2(**nondiffable2)
-        qml.Displacement(1., 1., wires=1)
-        return [qml.expval(qml.Identity(0)), qml.expval(qml.X(1))]
-
-    return circuit
-
-
-######################
-
-
-class TestIntegrationCircuit:
-    """Tests the integration of templates into circuits using different interfaces. """
-
-    @pytest.mark.parametrize("template1, diffable1, nondiffable1", QUBIT_DIFFABLE_NONDIFFABLE)
-    @pytest.mark.parametrize("template2, diffable2, nondiffable2", QUBIT_DIFFABLE_NONDIFFABLE)
+# List templates in NO_OP_BEFORE that do not allow for operations
+# before they are called in a quantum function.
+# These templates will be skipped in tests of that nature.
+
+NO_OP_BEFORE = [qml.templates.AmplitudeEmbedding, qml.templates.BasisEmbedding]
+
+# Each entry to QUBIT_INIT and CV_INIT adds a template with specified inputs to the
+# integration tests ``TestIntegrationInitFunctions``
+
+# The entries have the following form:
+#
+# (template, dict of differentiable arguments, dict of non-differentiable arguments)
+#
+# The dictionary of differentiable arguments calls the initialization function.
+
+QUBIT_INIT = [(qml.templates.StronglyEntanglingLayers,
+               {'weights': qml.init.strong_ent_layers_uniform(n_layers=3, n_wires=2), 'wires': range(2)}),
+              (qml.templates.StronglyEntanglingLayers,
+               {'weights': qml.init.strong_ent_layers_uniform(n_layers=2, n_wires=3), 'wires': range(3)}),
+              (qml.templates.StronglyEntanglingLayers,
+               {'weights': qml.init.strong_ent_layers_normal(n_layers=3, n_wires=2), 'wires': range(2)}),
+              (qml.templates.StronglyEntanglingLayers,
+               {'weights': qml.init.strong_ent_layers_normal(n_layers=2, n_wires=3), 'wires': range(3)}),
+              (qml.templates.RandomLayers,
+               {'weights': qml.init.random_layers_uniform(n_layers=3, n_rots=2, n_wires=2), 'wires': range(2)}),
+              (qml.templates.RandomLayers,
+               {'weights': qml.init.random_layers_uniform(n_layers=3, n_rots=2, n_wires=2), 'wires': range(2)}),
+              (qml.templates.RandomLayers,
+               {'weights': qml.init.random_layers_normal(n_layers=2, n_rots=2, n_wires=3), 'wires': range(3)}),
+              (qml.templates.RandomLayers,
+               {'weights': qml.init.random_layers_normal(n_layers=2, n_rots=2, n_wires=3), 'wires': range(3)}),
```

Comment:
There seems to be a duplication here (2-2 being pairwise duplicates), is that intentional?

Reply:
Well spotted!!

I think that was a classic "copy and paste, but forgot to change". Fixed.